src/main/java/org/sciborgs1155/robot/pivot/Pivot.java

```diff
@@ -0,0 +1,160 @@
+
+  /** Creates a real or simulated pivot based on {@link Robot#isReal()} */
+  public static Pivot create() {
+    return Robot.isReal() ? new Pivot(new RealPivot()) : new Pivot(new SimPivot());
+  }
+
+  public Pivot(PivotIO pivot) {
+    this.pivot = pivot;
+    sysIdRoutine =
+        new SysIdRoutine(
+            new SysIdRoutine.Config(),
+            new SysIdRoutine.Mechanism(
+                (v) -> pivot.setVoltage(v.in(Volts)), null, this, "pivotSysId"));
+
+    SmartDashboard.putData("quasistaticForward", quasistaticForward());
+    SmartDashboard.putData("quasistaticBack", quasistaticBack());
+    SmartDashboard.putData("dynamicForward", dynamicForward());
+    SmartDashboard.putData("dynamicBack", dynamicBack());
+  }
+
+  /**
+   * Smoothly angle the pivot to a desired goal using a {@link ProfiledPIDController}.
+   *
+   * @param goalAngle The position to move the pivot to.
+   * @return The command to set the pivot's angle.
+   */
+  public Command runPivot(Supplier<Rotation2d> goalAngle) {
+    return runOnce(() -> pivotPID.setGoal(goalAngle.get().getRadians()))
+        .andThen(
+            run(() ->
+                    pivot.setVoltage(
+                        pivotPID.calculate(pivot.getPosition().getRadians())
+                            + pivotFeedforward.calculate(
+                                pivotPID.getSetpoint().position, pivotPID.getSetpoint().velocity)))
+                .withName("running Pivot"));
+  }
+
+  public Command manualPivot(Supplier<Double> joystick) {
+    return run(
+        () -> {
+          double velocity = joystick.get() * PivotConstants.MAX_VELOCITY.in(RadiansPerSecond);
+          double periodMovement = Constants.PERIOD.in(Second) * velocity;
+          double draftSetpoint = periodMovement + pivot.getPosition().getRadians();
+          double setpoint =
+              Math.max(Math.min(MAX_ANGLE.in(Radians), draftSetpoint), MIN_ANGLE.in(Radians));
+          pivot.setVoltage(
+              pivotPID.calculate(pivot.getPosition().getRadians(), setpoint)
+                  + pivotFeedforward.calculate(setpoint, 0));
+        });
+  }
+
+  public Command climb(Supplier<Rotation2d> goalAngle) {
+    return runOnce(() -> climbPID.setGoal(goalAngle.get().getRadians()))
+        .andThen(
+            run(() ->
+                    pivot.setVoltage(
+                        climbPID.calculate(pivot.getPosition().getRadians())
+                            + climbFeedforward.calculate(
+                                climbPID.getSetpoint().position, climbPID.getSetpoint().velocity)))
+                .withName("running Climb"));
+  }
+
+  @Log.NT
+  public Rotation2d getPosition() {
+    return pivot.getPosition();
+  }
+
+  @Log.NT
+  public boolean atSetpoint() {
+    return pivotPID.atSetpoint();
+  }
+
+  // ProfilePID doesn't log this stuff
+  @Log.NT
+  private double setpointRadians() {
+    return pivotPID.getSetpoint().position;
+  }
+
+  public Command quasistaticForward() {
```

Comment:
we should add position constraints to these commands, to avoid the pivot from destroying itself
for example
```java
sysIdRoutine.quasistatic(Direction.kForward).until(() -> pivot.getPosition.getRadians() > MAX_ANGLE)
```